clang/lib/AST/RecordLayoutBuilder.cpp

```diff
@@ -2199,8 +2199,9 @@ static bool mustSkipTailPadding(TargetCXXABI ABI, const CXXRecordDecl *RD) {
   llvm_unreachable("bad tail-padding use kind");
 }
 
-static bool isMsLayout(const ASTContext &Context) {
-  return Context.getTargetInfo().getCXXABI().isMicrosoft();
+static bool isMsLayout(const ASTContext &Context, bool CheckAuxABI = false) {
```

Comment:
What if we change parameter type to `TargetInfo`?
```suggestion
static bool isMsLayout(const TargeInfo *TI) {
```
usage
```c++
if (isMsLayout(*this.getTargetInfo())) ...
...
if (isMsLayout(*this.getAuxTargetInfo())) ...
```

Reply:
Then we will need to change usage of this function for 3 additional times not including the one which I changed. And these 3 additional places of `isMsLayout` aren't connected with my current patch. Are you sure?

Reply:
I don't see any problems with this refactoring.
The benefit I see from this change is that we move `if` statement higher in the call stack as exactly the same logic is already implemented higher in the calls stack. On the other hand, I hope all `isMsLayout` calls are inlined and this might not be a problem.
I'll leave final decision to you.

Reply:
I'd like to keep unrelated parts of code unchanged.